### PR TITLE
Working testPubSub with ready for review.

### PR DESCRIPTION
**Pull request by debruce**
This has a test case for inter-thread and one for inter-process. Each is just point to point. Each does a thousand trials of random data messages of randomly generated length from 1 byte to 1400.

### CODE REVIEW

File: test/CMakeLists.txt
Comment (gregmedd) [inline test/CMakeLists.txt:56]: Minor: there's some indentation strangeness here.

```suggestion
add_executable(testPubSub
	src/testPubSub.cpp)
target_link_libraries(testPubSub
	PUBLIC
		up-client-zenoh-cpp::up-client-zenoh-cpp
		spdlog::spdlog
	PRIVATE
		GTest::gtest_main
		GTest::gmock    
		pthread
)
```

File: test/src/testPubSub.cpp
Comment (gregmedd) [inline test/src/testPubSub.cpp:28]: Minor: remove if unused
```suggestion
```

File: test/src/testPubSub.cpp
Comment (gregmedd) [inline test/src/testPubSub.cpp:55]: Minor: `using` style type definitions are more consistent with the rest of the code base.
```suggestion
    using Data = std::vector<std::uint8_t>;
```

File: test/src/testPubSub.cpp
Comment (gregmedd) [inline test/src/testPubSub.cpp:131]: Minor: predicate form of `wait_until` can be used here
```suggestion
                cv.wait_until(lock, end_time, [&count, &queue](){ count = queue.size(); return count >= 1; });
```

File: test/src/testPubSub.cpp
Comment (gregmedd) [inline test/src/testPubSub.cpp:106]: Minor: use `std::chrono` and chrono literals for time values
```suggestion
    std::tuple<size_t, Data, std::chrono::steady_clock::time_point>> get(std::chrono::milliseconds max_wait = 1s)
```

File: test/src/testPubSub.cpp
Comment (gregmedd) [inline test/src/testPubSub.cpp:1]: This does not look like a unit test to me. Rather, it looks more like a functional or integration test.

If that is the case, then we would want to keep it separate from the unit tests so that it does not inadvertently contribute to code coverage numbers (when that eventually gets implemented). At the very least, the test source should be in a different folder than unit tests.

If this is instead a unit test, please clarify which code it is intended to unit test.

**Review by gregmedd — CHANGES_REQUESTED** [review]
I think I'm ok with this change overall once comments are addressed. I would like to see a little more description of what is being tested and how the test is doing that.

**Comment by gregmedd** [comment]
One last note: please remember to rebase this PR to main once #23 is merged

File: test/src/testPubSub.cpp
Comment (debruce) [reply test/src/testPubSub.cpp:218]: It is a reminder of the fact that child process cleanup is not practical, or not worth implementing, and that child process errors cannot practically be directed back to the parent test. This could be fixed but there are bigger fish to fry, I think.

File: test/CMakeLists.txt
Comment (stevenhartley) [reply test/CMakeLists.txt:56]: Please reach out to Mark Stokes

File: test/src/testPubSub.cpp
Comment (gregmedd) [reply test/src/testPubSub.cpp:1]: I think we can add that as a discussion point on https://github.com/eclipse-uprotocol/up-cpp/issues/83 and close the thread here.

File: test/src/testPubSub.cpp
Comment (debruce) [reply test/src/testPubSub.cpp:28]: Will be resolved in next push.

File: test/src/testPubSub.cpp
Comment (debruce) [reply test/src/testPubSub.cpp:55]: Will be resolved in next push.

File: test/src/testPubSub.cpp
Comment (debruce) [reply test/src/testPubSub.cpp:106]: I will address this in next checkin, but I'm not thrilled with chrono literals for default argument values, since it seems to pollute namespace. Can we discuss this after next update?

File: test/src/testPubSub.cpp
Comment (debruce) [reply test/src/testPubSub.cpp:131]: Adopting this in next push.

File: test/src/testPubSub.cpp
Comment (billpittman) [reply test/src/testPubSub.cpp]: This still seems to be an open issue.

**Review by billpittman — APPROVED** [review]
LGTM.  Thanks for making the changes.

File: test/CMakeLists.txt
Comment (gregmedd) [inline test/CMakeLists.txt:131]: Is pthread added twice here?

File: test/src/testPubSub.cpp
Comment (gregmedd) [inline test/src/testPubSub.cpp:105]: Minor: replace std::chrono::time_point<...> with std::chrono::steady_clock::time_point

File: test/src/testPubSub.cpp
Comment (gregmedd) [inline test/src/testPubSub.cpp:212]: Minor: Use chrono literals

**Review by gregmedd — APPROVED** [review]
I've left a few more comments, but none of them are things I would hold up this change over. If you have a chance to fix them, great. Otherwise I'm ok merging this.

**Pull request merged**
